core/src/main/java/hudson/triggers/Trigger.java

```diff
@@ -254,7 +254,9 @@ public static void checkTriggers(final Calendar cal) {
                     public void run(AbstractProject p) {
                         for (Trigger t : (Collection<Trigger>) p.getTriggers().values()) {
                             if (t instanceof SCMTrigger) {
-                                LOGGER.fine("synchronously triggering SCMTrigger for project " + t.job.getName());
+                                if (t.job != null) {
+                                    LOGGER.fine("synchronously triggering SCMTrigger for project " + t.job.getName());
```

Comment:
I think spotbugs still complain because t.job could be non null on the if line, but it could be null on the LOGGER line. Shouldn't you assign t.job to a variable and use it on the LOGGER line to avoid the warning?